mirror.py

```diff
@@ -44,38 +44,26 @@ def sitewide_integration(cls, _db):
             )
             return None
 
-        if len(integrations) > 1:
-            # If there are multiple integrations configured, none of
-            # them can be the 'site-wide' configuration.
-            raise CannotLoadConfiguration(
-                'Multiple storage integrations are configured'
-            )
-
         [integration] = integrations
         return integration
 
     @classmethod
-    def for_collection(cls, collection, use_sitewide=False):
+    def for_collection(cls, collection):
         """Create a MirrorUploader for the given Collection.
 
         :param collection: Use the mirror configuration for this Collection.
 
-        :param use_sitewide: If there's no mirror for this specific Collection,
-            should we return a sitewide mirror instead?
-
         :return: A MirrorUploader, or None if the Collection has no
             mirror integration.
         """
         integration = collection.mirror_integration
         if not integration:
-            if use_sitewide:
-                try:
-                    from model import Session
-                    _db = Session.object_session(collection)
-                    return cls.sitewide(_db)
-                except CannotLoadConfiguration, e:
-                    return None
-            else:
+            try:
+                from model import Session
+                _db = Session.object_session(collection)
+                #todo get name of storage
+                return cls.mirror(_db)
```

Comment:
Here I'm not sure how to get the name of the storage integration because the collection alone is not enough to get the right implementation.

Reply:
Collection.mirror_integration will get you the ExternalIntegration that should be used to mirror stuff for a given collection. Then you can call MirrorUploader.implementation() on that ExternalIntegration.